src/Sylius/Bundle/ResourceBundle/Behat/DefaultContext.php

```diff
@@ -456,6 +456,37 @@ protected function replaceColumnNamesWithColumnIds(NodeElement $table, array $fi
     }
 
     /**
+     * Callback can return any value, to stop waiting use anonymous function provided as first argument.
+     *
+     * @param callable $callback
+     * @param int $limit
+     * @param int $delay In miliseconds
+     *
+     * @return mixed
+     *
+     * @throws \RuntimeException If timeout was reached
+     */
+    protected function waitFor(callable $callback, $limit = 10, $delay = 100)
+    {
+        $forceBreak = false;
+        $emergencyExit = function () use (&$forceBreak) {
+            $forceBreak = true;
+        };
+
+        for ($i = 0; $i < $limit; ++$i) {
+            $payload = $callback($emergencyExit);
+
+            if ($forceBreak) {
```

Comment:
Maybe we could simply use `null !== $payload` and get rid of this `emergencyExit`. It will make code more readable and less 'magic' :wink: `